frontend/src/components/Events/EventDetails.js

```diff
@@ -28,17 +31,27 @@ const Button1 = styled(Button)({
 })
 
 function EventDetails(props) {
+
+    const addParticipant = () => {
+        if (!JSON.stringify(props.event.participants).includes(JSON.stringify({uid: firebase.auth().currentUser.uid, email:firebase.auth().currentUser.email}))) {
```

Comment:
This is rather unreadable. When you have more complex conditionals, you should pull it out into a variable and give it a descriptive name. In this case, you could do call it something like `isUserInParticipantsList`.